src/unix/udp.c

```diff
@@ -629,12 +629,15 @@ int uv__udp_send(uv_udp_send_t* req,
      * write.
      */
     if (!uv__queue_empty(&handle->write_queue))
-      uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
+      err = uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
   } else {
-    uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
+    err = uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
   }
 
-  return 0;
+  if (err)
+    uv__handle_stop(handle);  /* The uv__io_start failed, undo handle op */
```

Comment:
This leaks `req->bufs` when `req->bufs != req->bufsml` and leaves `req` in `handle->write_queue`.

Reply:
Not anymore. :-)